### ISSUE TITLE
E206 VariableHasSpacesRule: Fix multiline nested JSON false positive 

### ISSUE THREAD

**Issue by simonkeyd** [ansible-community/ansible-lint#1671]
##### Summary
`ansible-lint` raises an E206 error when linting a nested JSON object spanned on multiple lines.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
```console
$ ansible --version
ansible 2.9.4

$ ansible-lint --version
ansible-lint 5.0.12 using ansible 2.9.4
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT
Running in a WSL1 environment:
```console
$ uname -r
4.4.0-18362-Microsoft
$ cat /etc/debian_version
10.9
$ python --version
Python 3.7.3
```

##### STEPS TO REPRODUCE
* Using the following command generate a matching false positive playbook:
```bash
cat << EOF > e206-nested-json.yml
- hosts: localhost
  connection: local
  gather_facts: false
  tasks:
    - name: Valid multiline nested JSON false positive
      debug:
        msg: >-
          {{ {'dummy_2': {'nested_dummy_1': value_1,
                          'nested_dummy_2': value_2}} |
          combine(dummy_1) }}
EOF
```
where problematic line being the following: `'nested_dummy_2': value_2}} |`
* Try to lint the freshly created playbook
```bash
ansible-lint -q e206-nested-json.yml
```

##### Desired Behaviour
No error should be raised as one is only closing a multiline nested JSON dictionary while using a valid syntax.

##### Actual Behaviour
The following error is raised by `ansible-lint`:
```console
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: e206-nested-json.yml
WARNING  Listing 1 violation(s) that are fatal
var-spacing: Variables should have spaces before and after: {{ {'dummy_2': {'nested_dummy_1': value_1,
                'nested_dummy_2': value_2}} |
combine(dummy_1) }}
e206-nested-json.yml:5 Task/Handler: Valid multiline nested JSON false positive

You can skip specific rules or tags by adding them to your configuration file:
```
And as stated above the following line is causing the issue:
`'nested_dummy_2': value_2}} |`
